be/src/exprs/aggregate_functions.cpp

```diff
@@ -1114,50 +1115,46 @@ StringVal AggregateFunctions::hll_finalize(FunctionContext* ctx, const StringVal
 }
 
     
-void AggregateFunctions::hll_union_agg_init(FunctionContext* ctx, StringVal* dst) {
-    int str_len = std::pow(2, HLL_PRECISION);
-    dst->is_null = false;
-    dst->ptr = ctx->allocate(str_len);
-    dst->len = str_len;
-    memset(dst->ptr, 0, str_len);
+void AggregateFunctions::hll_union_agg_init(FunctionContext* ctx, HllVal* dst) {
+    dst->init(ctx);
 }
 
-void AggregateFunctions::hll_union_parse_and_cal(HllSetResolver& resolver, StringVal* dst) {
+void AggregateFunctions::hll_union_parse_and_cal(HllSetResolver& resolver, HllVal* dst) {
     
```

Comment:
hll_union_parse_and_cal can be a member of HllVal， and it support a function that merge other type or src StringVal, so we will get a better encapsulation for hll.